server/src/main/java/org/opensearch/identity/ExtensionTokenProcessor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+
+/**
+ * Token processor class to handle token encryption/decryption
+ */
+public class ExtensionTokenProcessor {
+    private String extensionUniqueId;
+
+    public ExtensionTokenProcessor(String extensionUniqueId) {
+        this.extensionUniqueId = extensionUniqueId;
+    }
+
+    public String getExtensionUniqueId() {
+        return extensionUniqueId;
+    }
+
+    /**
+     * ENCRYPTION
+     *
+     * Create a two-way encrypted access token for given principal for this extension
+     * @return token generated from principal
+     */
+    public PrincipalIdentifierToken generateToken(Principal principal) {
+        String token = principal.getName() + ":" + extensionUniqueId;
+        return new PrincipalIdentifierToken(token);
+    }
+
+    /**
+     * DECRYPTION
+     *
+     * Convert token to Principal
+     * @return Principal
+     */
+    public Principal extractPrincipal(PrincipalIdentifierToken token) {
+        String[] parts = token.getToken().split(":");
+
+        if (parts.length != 2) {
```

Comment:
While this matches the current placeholder implementation this isn't the final decryption. Need a lot more comments here explaining the future plan for encrypt/decrypt here.

Reply:
@DarshitChanpura Could you create issue(s) and then link them in here?

Reply:
Sure. [The issue](https://github.com/opensearch-project/OpenSearch/issues/4485) is already created. I will add them here.